### PR TITLE
[Feature] Support using FlexKV as anothor KV Cache Offloading option.

### PR DESCRIPTION
**Pull request by axxx03**
# Description
FlexKV is a distributed KV store and multi-level cache management system developed by Tencent Cloud's TACO team in collaboration with the community, designed for large-scale LLM inference scenarios. FlexKV leverages multi-level caching to enable inference engines to achieve higher throughput and lower latency.

In our case, when intergated with FlexKV, we can achieve the following improvement:

- ISL=21K, OSL=1K, batch_size=8: TTFT decreases by 60%, TPOT increases by 13%, and QPM increases by 16%.

### CODE REVIEW

File: vllm/distributed/kv_transfer/kv_connector/v1/flexkv_connector.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+from typing import TYPE_CHECKING, Any, Optional
+
+import torch
+from flexkv.integration.vllm.vllm_v1_adapter import FlexKVConnectorV1Impl
```
Comment (chaunceyjiang) [inline vllm/distributed/kv_transfer/kv_connector/v1/flexkv_connector.py:6]: Just a question: is flexkv an open-source project?

Comment (axxx03) [reply vllm/distributed/kv_transfer/kv_connector/v1/flexkv_connector.py:6]: Yes, our repository is available here: https://github.com/taco-project/FlexKV￼

Comment (KuntaiDu) [reply vllm/distributed/kv_transfer/kv_connector/v1/flexkv_connector.py:6]: I clicked this link and it gives me a 404

**Review by KuntaiDu — CHANGES_REQUESTED** [review]
Ideally we shouldn't change the scheduler when adding new connector.

File: vllm/v1/core/sched/scheduler.py [hunk 2]
```diff
@@ -164,6 +157,7 @@ def __init__(
         # KV Connector: requests in process of async KV loading or recving
         self.finished_recving_kv_req_ids: set[str] = set()
         self.failed_recving_kv_req_ids: set[str] = set()
+        self.sending_kv_reqs: dict[str, Request] = {}
```
Comment (KuntaiDu) [inline vllm/v1/core/sched/scheduler.py]: Just wondering the motivation of adding this `sending_kv_reqs`

File: vllm/v1/core/sched/scheduler.py [hunk 3]
```diff
@@ -1649,6 +1580,10 @@ def _update_from_kv_xfer_finished(self, kv_connector_output: KVConnectorOutput):
             schedule the request during the next step.
         """
 
+        # avoid busy checking
+        if len(self.running) == 0:
+            time.sleep(0.01)
```
Comment (KuntaiDu) [inline vllm/v1/core/sched/scheduler.py]: Just wondering why busy checking is not ideal here? (scheduler process don't have any other things to do anyway)

File: vllm/distributed/kv_transfer/kv_connector/v1/flexkv_connector.py (hunk 1 above)
Comment (axxx03) [reply vllm/distributed/kv_transfer/kv_connector/v1/flexkv_connector.py:6]: Sorry, the link above contains an extra Unicode character. Please use this:
https://github.com/taco-project/FlexKV

File: vllm/v1/core/sched/scheduler.py (hunk 3 above)
Comment (axxx03) [reply vllm/v1/core/sched/scheduler.py]: Thanks, I'll remove this.

File: vllm/v1/core/sched/scheduler.py (hunk 2 above)
Comment (axxx03) [reply vllm/v1/core/sched/scheduler.py]: Thanks for the review.

Here is the motivation: In KV cache offloading scenario, requests are offloaded only after decode finishes. Therefore, for the final request, when it start offloading KV cache, the scheduler may think there’s no ongoing task and switch to idle state. So, `update_from_output` wouldn’t be called for that task to refresh its state. 

After review the code, we find it’s OK to remove `sending_kv_reqs ` and let the task state be refreshed when the next request arrives.

Therefore, we will remove this.
